### PR TITLE
[C-2448] Fix common dependencies and lint

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Fixes dependencies and lint issues in common due to dev/peer/dependency setup. 
- Also adds an @audius/common typecheck + lint step to ensure common commits pass eslint
- Adds react and react-redux to common devDeps and peerDeps, and updates mobile metro.config.js to resolve local versions to prevent the "multiple instances of react" error
- Update audius-eslint-config to use typescript-resolver explictly

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2448-fix-common-lint

File: .circleci/src/commands/@commands.yml [hunk 1]
```diff
@@ -16,4 +16,12 @@ create_concatenated_patch_file:
   steps:
     - run:
         name: Combine patch-files to single file
-        command: ls -d -- packages/*/patches/*.patch | xargs cat > << parameters.filename >>
\ No newline at end of file
+        command: ls -d -- packages/*/patches/*.patch | xargs cat > << parameters.filename >>
+
+verify-common:
```
Comment (sliptype) [inline .circleci/src/commands/@commands.yml:21]: Could you put this in it's own `@common-commands.yml` file? Would be better to keep `@commands.yml` as only root level commands

**Review by sliptype — APPROVED** [review]
Awesome!

File: .circleci/src/commands/@commands.yml (hunk 1 above)
Comment (dylanjeffers) [reply .circleci/src/commands/@commands.yml:21]: yeah great idea

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2448-fix-common-lint

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Pull request merged**